test/test.go

```diff
@@ -41,3 +45,38 @@ func GetEnvVarWithFallback(t *testing.T, envVars ...string) string {
 
 	return envVarValue.(string)
 }
+func MapToTestChecks(fqrn string, fields map[string]interface{}) []resource.TestCheckFunc {
+	var result []resource.TestCheckFunc
+	for key, value := range fields {
+		switch reflect.TypeOf(value).Kind() {
+		case reflect.Slice:
+			for i, lv := range value.([]interface{}) {
+				result = append(result, resource.TestCheckResourceAttr(
+					fqrn,
+					fmt.Sprintf("%s.%d", key, i),
+					fmt.Sprintf("%v", lv),
+				))
+			}
+		case reflect.Map:
+			// this also gets generated, but it's value is '1', which is also the size. So, I don't know
+			// what it means
+			// content_synchronisation.0.%
+			resource.TestCheckResourceAttr(
+				fqrn,
+				fmt.Sprintf("%s.#", key),
```

Comment:
The end of this [section](https://www.terraform.io/plugin/sdkv2/testing/acceptance-tests/teststep#builtin-check-functions) in TF doc shows for `map`, `%` is used instead of `#` for number of keys.